### PR TITLE
add error loc to range2

### PR DESCRIPTION
**Pull request by shawnl**
This method (one page at a time) does not decrease the performance.

### CODE REVIEW

**Comment by shawnl** [comment]
Also switched to aligned loads.

File: range2-sse.c [hunk 1]
```diff
@@ -7,58 +7,77 @@
...
+          return r;
+        data += len_to_aligned;
+        len -= len_to_aligned;
+
         __m128i prev_input = _mm_set1_epi8(0);
         __m128i prev_first_len = _mm_set1_epi8(0);
 
         const __m128i first_len_tbl =
-            _mm_lddqu_si128((const __m128i *)_first_len_tbl);
+            _mm_loadu_si128((const __m128i *)_first_len_tbl);
```
Comment (cyb70289) [inline range2-sse.c]: should it be _mm_load_si128?

**Comment by cyb70289** [comment]
Thanks @shawnl for your patch.
I see big performance drop of range2 code (run "./utf8 bench") from 5440 to 1428 MB/s on my i7-4790 host. Would you double check?

File: range2-sse.c (hunk 1 above)
Comment (shawnl) [reply range2-sse.c]: oh yes.

**Comment by shawnl** [comment]
bench range2... pass
    time: 0.2253 s
    data: 1023 MB
    BW: 4542.26 MB/s

**Comment by shawnl** [comment]
Maybe if performs bad on certain CPUs....

**Comment by cyb70289** [comment]
Thanks @shawnl, I will do more tests and analysis.

**Comment by shawnl** [comment]
What compiler are you using? I get the same results with gcc 8 and clang 8, but that compound check could be getting compiler poorely.

**Comment by cyb70289** [comment]
I tested with gcc 7 and 8.

**Comment by shawnl** [comment]
I just realized this actually isn't the right feature, because when printing the file it is desired to print the line number, which already requires scanning the whole file again (and with a slower algorithm).

**Pull request closed without merging**